components/script/dom/htmlformelement.rs

```diff
@@ -258,8 +262,83 @@ impl HTMLFormElementMethods for HTMLFormElement {
 
     // https://html.spec.whatwg.org/multipage/#the-form-element%3Adetermine-the-value-of-a-named-property
     fn NamedGetter(&self, name: DOMString) -> Option<RadioNodeListOrElement> {
-        // return Option::Some::<RadioNodeListOrElement>;
-        unimplemented!();
+        // unimplemented!();
+        let window = window_from_node(self);
+
+        // Q2. ERROR in the following declaration - check the attached screenshot (image2)
+        let mut candidates: Vec<Dom<HTMLElement>> = Vec::new();
+
+        // let mut candidates: RadioNodeList = RadioNodeList::new(&window, Dom<HTMLElement>);
+
+        let controls = self.controls.borrow();
+
+        for child in controls.iter() {
+            if child
+                .downcast::<HTMLElement>()
+                .map_or(false, |c| c.is_listed_element())
+            {
+                if child.has_attribute(&local_name!("id")) ||
+                    (child.has_attribute(&local_name!("name")) &&
+                        child.get_string_attribute(&local_name!("name")) == name)
+                {
+                    // println!("{:?}", child.what_is_this());
+                    candidates.push(Dom::from_ref(&*child.downcast::<HTMLElement>().unwrap()));
+                }
+            }
+        }
+
+        if candidates.len() == 0 {
+            for child in controls.iter() {
+                if child.is::<HTMLImageElement>() {
+                    if child.has_attribute(&local_name!("id")) ||
+                        (child.has_attribute(&local_name!("name")) &&
+                            child.get_string_attribute(&local_name!("name")) == name)
+                    {
+                        candidates.push(Dom::from_ref(&*child.downcast::<HTMLElement>().unwrap()));
+                    }
+                }
+            }
+        }
+
+        let mut past_names_map = self.past_names_map.borrow_mut();
+
+        if candidates.len() == 0 {
+            for (key, val) in past_names_map.iter() {
+                if *key == name {
+                    // println!("{:?}", val.what_is_this());
+                    return Some(RadioNodeListOrElement::Element(DomRoot::from_ref(
+                        &*val.upcast::<Element>(),
+                    )));
+                }
+            }
+        }
```

Comment:
We'll need to return None here if there is no matching entry in past_names_map.

Reply:
Handled.